container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,453 @@
+	Ports              []string
+	Volumes            []string
+	EntrypointPrefix   []string
+	Env                []string
+	Command            []string
+	Entrypoint         []string
+	Layer              []string
+	StampInfoFile      []string
+}
+
+// emptySHA256Digest returns the sha256 sum of an empty string.
+func emptySHA256Digest() (empty string) {
+	b := sha256.Sum256([]byte(""))
+	empty = hex.EncodeToString(b[:])
+	return
+}
+
+// extractValue returns the contents of a file pointed to by value if it starts with a '@'.
+func extractValue(value string) (string, error) {
+	if strings.HasPrefix(value, "@") {
+		f, err := os.Open(value[1:])
+		// f, err := os.Open(value)
+		defer f.Close()
+		shaHash, err := ioutil.ReadAll(f)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to read content from file %s", value)
+		}
+		return string(shaHash), nil
+	}
+	return value, errors.Wrapf(nil, "unexpected value, got: %s want: @{...}", value)
+}
+
+// keyValueToMap converts an array of strings separated by '=' into a map of key-value pairs.
+// if toFormat is set to True, adds a . in front of the value so that the string can be formatted.
+func keyValueToMap(value []string) map[string]string {
+	convMap := make(map[string]string)
+	var temp []string
+	for _, kvpair := range value {
+		temp = strings.Split(kvpair, "=")
+		key, val := temp[0], temp[1]
+		convMap[key] = val
+	}
+	return convMap
+}
+
+type formattedString map[string]interface{}
+
+// formateWithMap takes all variables of format {{.VAR}} in the input string `format`
+// and replaces it according to the map of parameters to values in `params`.
+func formatWithMap(format string, params formattedString) string {
+	msg := &bytes.Buffer{}
+	template.Must(template.New("").Parse(format)).Execute(msg, params)
+	return msg.String()
+}
+
+// Stamp provides the substitutions of variables inside {} using info in file pointed to
+// by stampInfoFile.
+func Stamp(inp string, stampInfoFile []string) (string, error) {
+	if len(stampInfoFile) == 0 || inp == "" {
+		return inp, nil
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character by default.
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+	// do string manipulation in order to mimic python string format.
+	// specifically, replace '{' with '{{.' and '}' with '}}'.
+	inpReformatted := strings.ReplaceAll(inp, "{", "{{.")
```

Comment:
This is too error prone. I'm not sure but I think `{` could be used in contexts other than stamp variables and these replacements could corrupt those. Instead of doing this I suggest:
```
o := inp
for k, v := range formatArgs {
   o = strings.ReplaceAll(o, fmt.Sprintf("{%s}", key), value)
}
```
This should allow you to deprecate the type 'formattedString` and the function `formatWithMap`